packages/ui5-middleware-livetranspile/lib/livetranspile.js

```diff
@@ -24,60 +24,71 @@ fileNotFoundError.file = ""
  * @returns {function} Middleware function to use
  */
 module.exports = function ({ resources, options }) {
-    const plugins =
-        options.configuration && options.configuration.transpileAsync
-            ? [
+    const config = options.configuration || {}
+    const plugins = config.transpileAsync
+        ? [
+              [
+                  "babel-plugin-transform-async-to-promises",
+                  {
+                      inlineHelpers: true
+                  }
+              ]
+          ]
+        : []
+    const babelConfig = config.babelConfig
+        ? config.babelConfig
+        : {
+              sourceMaps: "inline",
+              plugins,
+              presets: [
                   [
-                      "babel-plugin-transform-async-to-promises",
+                      "@babel/preset-env",
                       {
-                          inlineHelpers: true
+                          targets: {
+                              browsers: "last 2 versions, ie 10-11"
+                          }
                       }
                   ]
               ]
-            : []
-    const babelConfig =
-        options.configuration && options.configuration.babelConfig
-            ? options.configuration.babelConfig
-            : {
-                  sourceMaps: "inline",
-                  plugins,
-                  presets: [
-                      [
-                          "@babel/preset-env",
-                          {
-                              targets: {
-                                  browsers: "last 2 versions, ie 10-11"
-                              }
-                          }
-                      ]
-                  ]
-              }
-
+          }
+    const filePatternConfig = config.filePattern || ".js"
     return (req, res, next) => {
         if (
             req.path &&
             req.path.endsWith(".js") &&
```

Comment:
shouldn't this also be sth like `filePatternConfig.forEach(ending => req.path.endsWith(ending))` in order to act on all configured file endings? haven't validated, just by looking at the code.

Reply:
Actually it is not necessary, because req.path comes from the frontend calls to the node.js server. And the frontend only knows about js files. I cannot imagine a use case where the frontend wants to get the non-transpiled files directly... maybe just a lack of imagination, but a different feature request for sure :grinning: